presto-main/src/main/java/io/prestosql/sql/planner/EqualityInference.java

```diff
@@ -244,32 +244,30 @@ Expression getScopedCanonical(Expression expression, Predicate<Symbol> symbolSco
         if (canonicalIndex == null) {
             return null;
         }
-        return getCanonical(filter(equalitySets.get(canonicalIndex), symbolToExpressionPredicate(symbolScope)));
+        return getCanonical(filter(equalitySets.get(canonicalIndex), equivalentExpression -> isScoped(equivalentExpression, symbolScope)));
     }
 
-    private static Predicate<Expression> symbolToExpressionPredicate(final Predicate<Symbol> symbolScope)
+    private static boolean isScoped(Expression expression, Predicate<Symbol> symbolScope)
     {
-        return expression -> Iterables.all(SymbolsExtractor.extractUnique(expression), symbolScope);
+        return Iterables.all(SymbolsExtractor.extractUnique(expression), symbolScope);
```

Comment:
```
SymbolsExtractor.extractUnique(expression).stream()
  .allMatch(symbolScope);
```

would be good to remove Guava's `filter` too

Reply:
Good point. I'll do that in a separate change.